### ISSUE TITLE
[Products] Epic: Module Products

### ISSUE THREAD

**Issue by nielsdrost7** [InvoicePlane/InvoicePlane-v2#8]

**Comment by nielsdrost7** [comment]
## SMART Plan — Products module epic status

> **Context/Epic**: Products
>
> This is an epic/tracking issue, not a single feature — it aggregates the sub-issues below. The Measurable section tracks sub-issue completion rather than fabricating tests for the epic itself; each sub-issue gets its own SMART Plan when it's processed individually.

### Specific
Track delivery of the Products module: catalogue CRUD (done) plus the planned expansions — duplication, CSV import, attributes, stock management (tracking + invoice-finalisation deduction + permission enforcement), and profit margin.

### Measurable
- [x] `ProductResource`, `ProductCategoryResource`, `ProductUnitResource` shipped with CRUD test coverage in `Modules/Products/Tests/Feature/`.
- [ ] Partial import/export scaffolding — needs #140 (CSV import) to close it out.
- [ ] #271 — Duplicate a product — not started.
- [ ] #476 — Product attributes (Length, Width, etc.) — not started.
- [ ] #477 — Stock management (track quantities) — not started.
- [ ] #478 — Profit margin on products — not started.
- [ ] #66 — Auto-deduct stock on invoice finalisation — not started; depends on #477.
- [ ] #486 — Stock management linked to Invoices/Quotes — not started; depends on #477.
- [ ] #496 — Permission enforcement for Products module — not started.

**Definition of done for this epic**: all sub-issues above are closed, `Modules/Products/Tests/Feature/` covers each new capability, and the "Out of scope" list below is confirmed still out of scope at epic close.

### Achievable
1. Sequence stock-dependent work after #477 (stock management) lands, since #66 and #486 both depend on it.
2. #271 (duplicate) and #476 (attributes) have no dependencies and can be picked up independently.
3. #496 (permission enforcement) should land last, once the surface area it needs to guard (attributes, stock, margin) is stable.

### Relevant
The Products catalogue is a dependency for both Invoices and Quotes line items — incomplete stock/attribute support here blocks downstream features in those modules.

### Time-bound
No fixed date — epic closes when its sub-issues close. Re-review at each sprint boundary.

### Out of scope (unchanged)
- Product images in PDF line items
- Supplier/purchasing integration
